API.md

```diff
@@ -36,13 +36,13 @@ And a CODAP specific API that was also used for the initial version of the [Clou
 
 The second version of the API is handled by the documents_v2 controller and exposes only a non-GUI programmatic API in the /v2/ namespace as:
 
-- GET /v2/documents/:id - returns the contents of a document specified by id.  This requires either the document's read-only or read-write access keys.
+- GET /v2/documents/:id - returns the contents of a document specified by id.  If the document is not shared this requires either the document's read-only or read-write access keys.
 
 - PUT /v2/documents/:id - updates the contents of a document specified by id.  This requires the document's read-write access keys.
 
 - PATCH /v2/documents/:id - updates the contents of a document specified by id using the JSON Patch specification.  This requires the document's read-write access keys.
 
-- POST /v2/documents - creates a copy of a document identified by the source query parameter and returns the new id of the document along with its read-only and read-write access keys.  The source document must be a shared document or else the copy with fail.  No access key is needed.
+- POST /v2/documents - either creates a new anonymous shared document using the raw post content or creates a copy of a document identified by the source query parameter.  The source document must be a shared document or else the copy with fail.  No access key is needed.  If a document is created this returns a 201 status code with the following json response:  ```{status: "[Created|Copied]", valid: true, id: <document.id>, readAccessKey: <document.read_access_key>, readWriteAccessKey: <document.read_write_access_key>}```.
```

Comment:
From an API point of view it seems strange that in once case the new document will be shared and in the other case it will not.  Technically the API only needs to support those two options. But I think it would be more clear if we required a `shared=true` param in order to create a shared document. This would allow someone to make non shared documents from scratch. And it would also allow users to make copies of existing documents which are shared.  We don't really need to support that, but I don't think it would hurt.  Hopefully it would make the API more clear. @dougmartin and @kswenson what do you think?

I also wonder if we should reconsider the term 'shared' here. It doesn't really mean the document has been shared with anyone. What it really means is that the document is publicly accessible. `public` might be a better name (or `is_public` if the is_ convention is already used in other places).  What do you think? Should we just stick with shared because that is already engrained?